### PR TITLE
Type time

### PR DESCRIPTION
**Pull request by ruudboon**
Rebased upon request https://github.com/phalcon/cphalcon/pull/12293 
@sergeyklay Looked at the other added Column types but didn't see any changed test. What test do you want me to change?

### CODE REVIEW

**Comment by sergeyklay** [comment]
I would like you to create a small test ([see example here](https://github.com/phalcon/cphalcon/blob/3.0.x/unit-tests/DbDialectTest.php)) using this namespace https://github.com/phalcon/cphalcon/tree/master/tests/unit/Db/Column

File: phalcon/db/adapter/pdo/mysql.zep [hunk 1]
```diff
@@ -118,6 +118,11 @@ class Mysql extends PdoAdapter
 				 * Special type for datetime
 				 */
 				let definition["type"] = Column::TYPE_DATETIME;
+			} elseif memstr(columnType, "time") {
+				/**
+                                 * Special type for time
+                                 */
+                                let definition["type"] = Column::TYPE_TIME;
```
Comment (sergeyklay) [inline phalcon/db/adapter/pdo/mysql.zep]: Could you please fix indents

File: phalcon/db/adapter/pdo/postgresql.zep [hunk 2]
```diff
@@ -158,6 +158,11 @@ class Postgresql extends PdoAdapter
 				 * Timestamp
 				 */
 				let definition["type"] = Column::TYPE_TIMESTAMP;
+			}elseif memstr(columnType, "time") {
+				/**
+                                 * Time
+                                 */
```
Comment (sergeyklay) [inline phalcon/db/adapter/pdo/postgresql.zep]: and here

**Review by sergeyklay — CHANGES_REQUESTED** [review]

File: phalcon/db/adapter/pdo/postgresql.zep (hunk 2 above)
Comment (ruudboon) [inline phalcon/db/adapter/pdo/postgresql.zep]: Sorry, missed that. Will fix it. Still working on those unit test btw. Try to push them today.

File: phalcon/db/dialect/mysql.zep [hunk 3]
```diff
@@ -98,7 +98,11 @@ class Mysql extends Dialect
 					let columnSql .= "TIMESTAMP";
 				}
 				break;
-
+			case Column::TYPE_TIME:
+                                if empty columnSql {
+                                        let columnSql .= "TIME";
+                                }
+                                break;
```
Comment (sergeyklay) [inline phalcon/db/dialect/mysql.zep]: there still issue with indents

File: phalcon/db/dialect/postgresql.zep [hunk 4]
```diff
@@ -91,10 +91,14 @@ class Postgresql extends Dialect
 					let columnSql .= "TIMESTAMP";
 				}
 				break;
-
 			case Column::TYPE_TIMESTAMP:
+                                if empty columnSql {
+                                        let columnSql .= "TIMESTAMP";
+                                }
+                                break;
```
Comment (sergeyklay) [inline phalcon/db/dialect/postgresql.zep]: and here

**Review by sergeyklay — CHANGES_REQUESTED** [review]

**Comment by sergeyklay** [comment]
Could you provide a solution for the SQLite. I mean converting TIME to date type, etc

**Comment by ruudboon** [comment]
I think everything is in. If there's more needed please let me know.

File: phalcon/db/dialect/sqlite.zep [hunk 5]
```diff
@@ -90,7 +90,11 @@ class Sqlite extends Dialect
 					let columnSql .= "DATETIME";
 				}
 				break;
-
+			case Column::TYPE_TIME:
+				if empty columnSql {
+					let columnSql .= "TIME";
```
Comment (sergeyklay) [inline phalcon/db/dialect/sqlite.zep]: Are you sure about this? https://www.sqlite.org/datatype3.html#date_and_time_datatype

That's why I asked you to provide a workaround. That is why we are not in a hurry to take all possible data types.

**Review by sergeyklay — CHANGES_REQUESTED** [review]

File: tests/unit/Db/Column/MysqlTest.php [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+ */
+class MysqlTest extends UnitTest
+{
+    public function testDetectTimeColumn()
+    {
+        $dialect = new MysqlDialect();
+        $column = new Column("time-test", array(
+            'type' => Column::TYPE_TIME
+        ));
+        $this->assertEquals($dialect->getColumnDefinition($column), 'TIME');
```
Comment (sergeyklay) [inline tests/unit/Db/Column/MysqlTest.php]: I'd like to see here a test for the table that has this data type. Please understand me correctly. We can't release framework with black box. 

For example:

```sql
CREATE TABLE `entries` (
  `id` INT PRIMARY KEY AUTO_INCREMENT,
  `start` TIME NOT NULL DEFAULT CURRENT_TIME,
  `end` TIME DEFAULT '23:59',
  `add` TIME NULL,
  `amount` INT NOT NULL,
  `price` INT NOT NULL
);
```

Could you please convert this into objects and back?

File: tests/unit/Db/Column/PostgresqlTest.php [hunk 7]
```diff
@@ -60,4 +61,13 @@ function () {
...
         );
     }
+
+    public function testDetectTimeColumn()
+    {
+        $dialect = new PostgresqlDialect();
+        $column = new Column("time-test", array(
+            'type' => Column::TYPE_TIME
+        ));
+        $this->assertEquals($dialect->getColumnDefinition($column), 'TIME');
```
Comment (sergeyklay) [inline tests/unit/Db/Column/PostgresqlTest.php]: See my comment for MysqlTest

File: tests/unit/Db/Column/SqliteTest.php [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+ */
+class SqliteTest extends UnitTest
+{
+    public function testDetectTimeColumn()
+    {
+        $dialect = new SqliteDialect();
+        $column = new Column("time-test", array(
+            'type' => Column::TYPE_TIME
+        ));
+        $this->assertEquals($dialect->getColumnDefinition($column), 'TIME');
```
Comment (sergeyklay) [inline tests/unit/Db/Column/SqliteTest.php]: See my comment for MysqlTest

**Review by sergeyklay — CHANGES_REQUESTED** [review]

File: tests/unit/Db/Column/MysqlTest.php (hunk 6 above)
Comment (ruudboon) [inline tests/unit/Db/Column/MysqlTest.php]: Hmm I get it. Makes sense! Shouldn't we have them for datetime etc also? 
Thnx 4 all the feedback. Appreciate the effort, kinda new to this.

Comment (Jurigag) [inline tests/unit/Db/Column/MysqlTest.php]: There is already `TYPE_DATETIME` and `TYPE_DATE`

**Comment by david-duncan** [comment]
This hasn't seen activity in two months - can we get review fixes pushed and for this PR to get squashed?

**Comment by ruudboon** [comment]
Currently looking into this (first need to setup a new dev env). 
Looks like the unit tests are changed a bit. Some pointer would be helpful.

**Comment by Jurigag** [comment]
You need to rebase it.

**Comment by ruudboon** [comment]
Just wanna give heads up that I did some work on this issue (https://github.com/ruudboon/cphalcon/commit/ce2408dca9fed7803b08ff33682c98d79e50c8b3). Trying to finish unit tests and get Travis to work on my branch before submitting a new pull.

**Comment by niden** [comment]
Addressed in https://github.com/phalcon/cphalcon/pull/13562

**Pull request closed without merging**